decoder.go

```diff
@@ -472,7 +462,9 @@ func (d *decoder) textUnmarshaller(key string, value reflect.Value, str string)
 		return nil
 	}
 
-	if value.IsNil() {
+	kind := value.Kind()
+
+	if kind >= reflect.Array && kind <= reflect.Slice && value.IsNil() {
```

Comment:
I mean at least a comment is required then, since it's completely non-obvious what's going on without jumping into the source of the kind definition